packages/api/src/brevo-contact/brevo-contact-import.controller.ts

```diff
@@ -33,23 +35,27 @@ export function createBrevoContactImportController({ Scope }: { Scope: Type<Emai
             }),
         )
         @RequiredPermission(["brevo-newsletter"], { skipScopeCheck: true })
-        async upload(@UploadedFile() file: Express.Multer.File, @Body("scope") scope: string, @Body("listIds") listIds?: string): Promise<void> {
-            const content = file.buffer.toString("utf8");
+        async upload(
+            @UploadedFile() file: Express.Multer.File,
+            @Body("scope") scope: string,
+            @Body("listIds") listIds?: string,
+        ): Promise<CSVImportInformation> {
             const parsedScope = JSON.parse(scope) as EmailCampaignScopeInterface;
-
             const redirectUrl = this.config.brevo.resolveConfig(parsedScope).redirectUrlForImport;
-            const contacts = await this.brevoContactImportService.parseCsvToBrevoContacts(content, redirectUrl);
-
-            if (contacts.length > 100) {
-                throw new CometValidationException("Too many contacts in file. Currently we only support 100 contacts at once.");
-            }
 
             let parsedListIds = undefined;
             if (listIds) parsedListIds = JSON.parse(listIds) as string[];
 
             const targetGroups = await this.targetGroupRepository.find({ id: { $in: parsedListIds } });
 
-            await this.brevoContactImportService.importContactsFromCsv(content, parsedScope, redirectUrl, targetGroups);
+            const stream = Readable.from(file.buffer);
```

Comment:
I have changed this to streaming, otherwise everything would be loaded into the memory.

The disadvantage is that I can't simply say that if the file has more than 100 contacts, it won't be treated at all.

But I think we should fix the problem that only 100 contacts can be imported via the admin anyway, otherwise the feature is probably a bit pointless.

Reply:
I agree, the admin import should be handled differently so more than 100 contacts can be imported. So this temporary solution is fine imo